docs/aws/audit/snsmonitoring/rules/is_topic_cmk_encrypted_remediation.mdx

```diff
@@ -1,6 +1,200 @@
+           sns_client.set_topic_attributes(
+               TopicArn=topic_arn,
+               AttributeName='KmsMasterKeyId',
+               AttributeValue='arn:aws:kms:your-region:your-account-id:key/your-key-id'
+           )
+           print(f"Updated topic {topic_arn} to use KMS CMK.")
+       else:
+           print(f"Topic {topic_arn} is already encrypted.")
+   ```
+
+4. **Run the Script**:
+   Execute the script to ensure all your SNS topics are encrypted using KMS CMKs.
+
+   ```bash
+   python encrypt_sns_topics.py
+   ```
+
+This script will iterate through all your SNS topics, check if they are encrypted, and if not, it will update them to use the specified KMS CMK. Make sure to replace `'arn:aws:kms:your-region:your-account-id:key/your-key-id'` with your actual KMS CMK ARN.
+</Accordion>
+
+</AccordionGroup>
+</Tab>
+<Tab title='Cause'>
+### Check Cause
+<AccordionGroup>
+<Accordion title='Using Console' defaultOpen='true'>
+1. Sign in to the AWS Management Console.
+2. Navigate to the Simple Notification Service (SNS) dashboard by selecting "Services" from the top menu, then typing "SNS" in the search box and selecting "Simple Notification Service".
+3. In the SNS dashboard, select "Topics" from the left-hand navigation pane. This will display a list of all the SNS topics available within your current AWS region.
+4. For each topic in the list, check the "KMS master key" column. If the column displays "Default", it means that the SNS topic is not encrypted with a customer-managed KMS key, indicating a potential misconfiguration.
+</Accordion>
+
+<Accordion title='Using CLI'>
+1. First, you need to list all the SNS topics in your AWS account. You can do this by using the following AWS CLI command:
+
+   ```
+   aws sns list-topics
+   ```
+   This command will return a list of all SNS topics in your AWS account.
+
+2. Once you have the list of SNS topics, you can check the encryption status of each topic. You can do this by using the following AWS CLI command:
+
+   ```
+   aws sns get-topic-attributes --topic-arn <TopicARN>
+   ```
+   Replace `<TopicARN>` with the ARN of the SNS topic you want to check. This command will return the attributes of the specified SNS topic.
+
+3. From the returned attributes, check the `KmsMasterKeyId` attribute. If this attribute is not present or is set to `alias/aws/sns`, then the SNS topic is not encrypted with a KMS CMK.
+
+4. Repeat steps 2 and 3 for each SNS topic in your AWS account.
+</Accordion>
+
+<Accordion title='Using Python'>
+1. Install and configure AWS SDK for Python (Boto3) on your local system. This will allow you to interact with AWS services including SNS.
+
+```python
+pip install boto3
+aws configure
+```
+
+2. Import the necessary modules and create a session using your AWS credentials.
+
+```python
+import boto3
+
+session = boto3.Session(
+    aws_access_key_id='YOUR_ACCESS_KEY',
+    aws_secret_access_key='YOUR_SECRET_KEY',
```

Comment:
<b>Secret:</b> Secret Keyword <br/>
                    <b><a href="https://console.cloudanix.com/code?t=secrets" target='_blank'>Click here for more details</a></b>